credit/metrics.py

```diff
@@ -100,3 +102,203 @@ def __call__(self, pred, y, clim=None, transform=None, forecast_datetime=0):
+
+        # DO NOT apply these weights during metrics computations, only on the loss during
+        self.w_var = None
+
+    def get_climatology(self, forecast_datetime, variable):
+        """Extract the climatology for the given forecast datetime and variable."""
+        forecast_datetime = datetime.utcfromtimestamp(forecast_datetime)
+        dayofyear = forecast_datetime.timetuple().tm_yday
+        hour = forecast_datetime.hour
+
+        # Extract climatology slice from xarray dataset
+        climatology_slice = self.climatology[variable].sel(
+            dayofyear=dayofyear, hour=hour, method="nearest"
+        )
+        # Convert to PyTorch tensor
+        return torch.tensor(climatology_slice.values, dtype=torch.float32)
+
+    def __call__(self, pred, y, transform=None, forecast_datetime=None):
+        if transform is not None:
+            pred = transform(pred)
+            y = transform(y)
+
+        # Get latitude and variable weights
+        w_lat = (
+            self.w_lat.to(dtype=pred.dtype, device=pred.device)
+            if self.w_lat is not None
+            else 1.0
+        )
+        w_var = (
+            self.w_var.to(dtype=pred.dtype, device=pred.device)
+            if self.w_var is not None
+            else 1.0
+        )
+
+        loss_dict = {}
+        with torch.no_grad():
+            # Compute ACC for acc_vars using anomalies
+            anamoly_scores = False
+            if self.climatology and forecast_datetime:
+                anomalies_pred = []
+                anomalies_y = []
+                acc_pred = pred
+                acc_y = y
+
+                # Get the list of variables from the climatology file
+                clim_vars = list(self.climatology.data_vars)
+
+                # Ensure self.acc_vars is in the same order as clim_vars
+                ordered_acc_vars = [var for var in clim_vars if var in self.vars]
+
+                # Reorder acc_pred and acc_y to match ordered_acc_vars
+                indices = [self.acc_vars.index(var) for var in ordered_acc_vars]
+                acc_pred = acc_pred[:, indices]
+                acc_y = acc_y[:, indices]
+
+                # Compute anamolies
+                for i, var in enumerate(ordered_acc_vars):
+                    clim = self.get_climatology(forecast_datetime, var).to(
+                        dtype=pred.dtype, device=pred.device
+                    ).unsqueeze(0)
+                    anomalies_pred.append(acc_pred[:, i] - clim)
+                    anomalies_y.append(acc_y[:, i] - clim)
+
+                anomalies_pred = torch.stack(anomalies_pred, dim=1)
+                anomalies_y = torch.stack(anomalies_y, dim=1)
+
+                for i, var in enumerate(self.acc_vars):
+                    pred_prime = anomalies_pred[:, i] - torch.mean(anomalies_pred[:, i])
+                    y_prime = anomalies_y[:, i] - torch.mean(anomalies_y[:, i])
+
+                    # Add epsilon to avoid division by zero
```

Comment:
I recommend using `torch.maximum(denominator, epsilon)` instead of adding epsilon to the denominator. 1e-7 may also be too large.

Reply:
Done -- I also changed epsilon to 1e-8